CMakeLists.txt

```diff
@@ -55,19 +94,37 @@ if(BUILD_TESTING)
   ament_add_gtest(pid_publisher_tests test/pid_publisher_tests.cpp)
   target_link_libraries(pid_publisher_tests control_toolbox)
   ament_target_dependencies(pid_publisher_tests rclcpp_lifecycle)
+
+  ## Control Filters
+  ament_add_gmock(test_low_pass_filter test/control_filters/test_low_pass_filter.cpp)
+  target_link_libraries(test_low_pass_filter low_pass_filter low_pass_filter_parameters)
+  ament_target_dependencies(test_low_pass_filter ${CONTROL_FILTERS_INCLUDE_DEPENDS})
+
+  ament_add_gmock(test_load_low_pass_filter test/control_filters/test_load_low_pass_filter.cpp)
+  target_link_libraries(test_load_low_pass_filter low_pass_filter low_pass_filter_parameters)
+  ament_target_dependencies(test_load_low_pass_filter ${CONTROL_FILTERS_INCLUDE_DEPENDS})
 endif()
 
+# Install
 install(
   DIRECTORY include/
   DESTINATION include/control_toolbox
 )
-install(TARGETS control_toolbox
+install(TARGETS control_toolbox low_pass_filter low_pass_filter_parameters
   EXPORT export_control_toolbox
   ARCHIVE DESTINATION lib
   LIBRARY DESTINATION lib
   RUNTIME DESTINATION bin
 )
 
+# Install pluginlib xml
+pluginlib_export_plugin_description_file(filters control_filters.xml)
+
 ament_export_targets(export_control_toolbox HAS_LIBRARY_TARGET)
-ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
+ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS} ${CONTROL_FILTERS_INCLUDE_DEPENDS})
+ament_export_libraries(
+  control_toolbox
+  low_pass_filter
+  low_pass_filter_parameters
+)
```

Comment:
This should not be necessary because we are exporting targets 2 lines above as they are defined in line 113-114.